microsoft-edge/devtools-guide-chromium/evaluate-performance/unminify.md

```diff
@@ -0,0 +1,106 @@
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.date: 03/08/2022
+---
+# Unminify performance profiles
+
+<!-- todo: review all mentions of "source mapping" in this article - what is the relation between these 3 features/technologies?
+*  source mapping and its UX/benefits
+*  unminification and its UX/benefits
+*  ADO secure server (for source maps) and its UX/benefits
+-->
+
+To see more meaningful original function names in the call stack profile in the **Performance** tool, instead of the transformed and minified function names, use the **Unminify** button.
+
+The minified call stack in the performance profile, with hard-to-read function names:
+
+![The minified call stack in the performance profile, with hard-to-read function names.](images/minified-perf-profile.msft.png)
+
+The unminified call stack in the performance profile, with some more meaningful function names restored:
+
+![The unminified call stack in the performance profile, with meaningful function names restored.](images/unminified-perf-profile.msft.png)
+
+When recording a performance profile in the **Performance** tool, the **Main** section of the profile displays detailed call stacks of JavaScript functions that were called during the recording.  
+
+Ideally, in this call stack view, you want to see the meaningful function names that are shown in your original source code.  However, if your production code is compiled, minified, or bundled, the **Performance** tool initially displays the function names that are extracted from your transformed, _production_ code.  These production function names might be obscure and might not easily map to function names in your original source code.
+
+Source mapping is used in DevTools to keep your client-side code readable and debuggable.  By clicking the **Unminify** button in the Performance tool, you can use source maps to create an unminified version of the performance profile you recorded.
+
+
+<!-- ====================================================================== -->
+## How to unminify a performance profile
+
+To unminify a performance profile, so that some more meaningful function names are shown in addition to the minified identifiers, follow the steps below.
+
+   **Set up source maps:**
+
+1. Set up source maps, to host your source maps side-by-side with your production code.  Host your source maps as well as your production code, where the **Performance** tool can find your source maps.<!-- Can you use ado server for secure hosting of source maps?-->  For details, see [Map the processed code to your original source code, for debugging](/microsoft-edge/devtools-guide-chromium/javascript/source-maps).
+
+   <!-- Or, if you are just trying out this feature, you can go to a browser-based Outlook mailbox: `https://outlook.office.com`, because this web app produces minified function names on the call stack. -->
+
+   In the following example, the source maps are served from the same server directory as the production code:
```

Comment:
@zoherghadyali - Which files in the listing .png are source map files?
Rendered view: https://review.docs.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/evaluate-performance/unminify?branch=pr-en-us-1784#how-to-unminify-a-performance-profile

Reply:
The navigator pane does not show source maps. It either shows the files downloaded from the web server (usually minified/uglified/bundled), or the original source files (which developers are familiar with). Source maps files are the underlying technology that makes it possible to go from the former to the latter, but are not displayed in devtools (apart from the Source Map Monitor tool).

Reply:
That being said, I think this line and the screenshot below it are confusing. This article talks about the performance tool, but here is a screenshot of the Sources tool, and we don't explain why, or what should we be looking at in the screenshot.

I think it might be better to remove those, and instead explain in details where to host source maps (next to "where the Performance tool can find your source maps", see my earlier comment).